### ISSUE TITLE
Getting ERROR while PDF render

### ISSUE THREAD

**Issue by igorlrgomes** [Syslifters/sysreptor#314]
I'm trying to render the PDF, but I always get this error message. Even using the demo report designs.

**Failed to load resource: net::ERR_NAME_NOT_RESOLVED in design "Demo Calzone v1.1"**

![image](https://github.com/user-attachments/assets/9d9fa9b7-d7c7-41b1-a8c2-8933a0fdca6e)

This is the docker logs sysreptor-app output:

2024-08-25 03:25:44,424 [INFO] root: POST /api/v1/auth/login/ 200 (user=reptor)
2024-08-25 03:25:44,540 [INFO] root: GET /api/v1/pentestusers/self/ 200 (user=reptor)
2024-08-25 03:25:44,676 [INFO] root: GET /api/v1/pentestusers/self/notifications/ 200 (user=reptor)
2024-08-25 03:25:46,856 [INFO] root: GET /api/v1/pentestprojects/?readonly=false&ordering=-created 200 (user=reptor)
2024-08-25 03:25:54,017 [INFO] root: GET /api/v1/projecttypes/?scope=global&ordering=name 200 (user=reptor)
2024-08-25 03:25:58,467 [INFO] root: GET /api/v1/projecttypes/4455e8f0-5b31-4c20-96dc-ff7b9f8751de/ 200 (user=reptor)
2024-08-25 03:25:58,615 [INFO] root: GET /api/v1/projecttypes/4455e8f0-5b31-4c20-96dc-ff7b9f8751de/ 200 (user=reptor)
2024-08-25 03:25:58,763 [INFO] root: POST /api/v1/projecttypes/4455e8f0-5b31-4c20-96dc-ff7b9f8751de/lock/ 201 (user=reptor)
2024-08-25 03:26:01,677 [INFO] root: Chromium messages: {ErrorMessage(level=<MessageLevel.ERROR: 'error'>, message='Failed to load resource: net::ERR_NAME_NOT_RESOLVED', details=None, location=None)}
2024-08-25 03:26:01,678 [INFO] root: Function render_to_html took 0:00:02.628801
2024-08-25 03:26:01,678 [INFO] root: Function render_pdf_task took 0:00:02.628994
2024-08-25 03:26:01,804 [INFO] root: POST /api/v1/projecttypes/4455e8f0-5b31-4c20-96dc-ff7b9f8751de/preview/ 200 (user=reptor)
2024-08-25 03:26:29,601 [INFO] root: POST /api/v1/projecttypes/4455e8f0-5b31-4c20-96dc-ff7b9f8751de/lock/ 200 (user=reptor)
2024-08-25 03:26:59,596 [INFO] root: POST /api/v1/projecttypes/4455e8f0-5b31-4c20-96dc-ff7b9f8751de/lock/ 200 (user=reptor)

Could someone help me?

**Comment by carter-falconops** [comment]
Same issue on our side. On all reports, we are getting that error.

**Comment by Maximator88** [comment]
I was just about to finish the report and after the update it is not possible to generate PDF. I am devastated.

"ERROR Failed to load resource: net::ERR_NAME_NOT_RESOLVED in design "Demo Calzone v1.1"

**Comment by zprytka** [comment]
Same error here, I cannot render any PDF of any report.

**Comment by MWedl** [comment]
Thanks for reporting. We will investigate the error and release an update ASAP

**Comment by MWedl** [comment]
TLDR: PDF rendering should work again, without updating SysReptor.

The problem arises from a update in chromium to version 128.0.6613.84.
This version is incompatible with playwright 1.45.1 resulting changed behavior when loading pages.

Chromium tries to load the URL https://pdf.sysreptor.com/favicon.ico. DNS queries to pdf.sysreptor.com fail (NXDOMAIN) because pdf.sysreptor.com is an internal dummy placeholder without actual DNS records.
This domain should never be loaded. However, since the latest chromium update, chromium tries to load data from pdf.sysreptor.com during PDF generation which results in PDF rendering errors.

As a temporary workaround we set up a webserver for https://pdf.sysreptor.com such that no `net::ERR_NAME_NOT_RESOLVED` errors occur anymore. Eventually you need to clear your DNS cache.

Playwright 1.46 fixes the root cause, such that chromium no longer tries to load data from https://pdf.sysreptor.com. We will update the library in the next release and also improve chromium network error handling.

Additionally, we are working on official docker images for SysReptor to prevent such errors in the future. Currently, all users have to build SysReptor docker images themselves during updates. Updates in system dependencies (such as the chromium debian package) result in different library versions (including potential breaking changes) for different users depending on the time they built the docker images. By providing official docker images, we are able to ensure that everyone uses the same set of dependencies that were tested before issuing the release.

**Comment by zprytka** [comment]
Yesterday it was working but today it still shows the same error. I did the workaround but still can't render the PDFs.

**Comment by Maximator88** [comment]
> Yesterday it was working but today it still shows the same error. I did the workaround but still can't render the PDFs.

I don't know why (maybe some cache) but I did 'update --force' and it worked again. I'm stressed because of all this. I have 36 hours left to make a report for the exam, and here I have such bad luck.

**Comment by aronmolnar** [comment]
@Maximator88 
For exams, you can use our [Cloud platform](https://docs.sysreptor.com/htb-reporting-with-sysreptor/) for free where you also have Professional features available.

You can also always send us your exported project to hello@syslifters.com if rendering doesn't work and we'll render it for you (or import it to the labs cloud).


> Yesterday it was working but today it still shows the same error. I did the workaround but still can't render the PDFs.

The error affects only SysReptor images that were built after Wednesday/Thursday last week.  
The error could still persist, if your SysReptor installation has no connection to the Internet. DNS caching shouldn't really be an issue because we set a new DNS record for pdf.sysreptor.com. The chromium browser then tries to access that URL via port 443 and load the favicon. If the container has no connection to the Internet, this connection and thus rendering will probably fail.

We will ship an update today that will resolve the issue permanently.

**Comment by aronmolnar** [comment]
Fixed in https://github.com/Syslifters/sysreptor/releases/tag/2024.69
